bin/node-template/README.md

```diff
@@ -128,98 +103,74 @@ Substrate-based blockchain nodes expose a number of capabilities:
 
 ### Pallets
 
-The runtime in this project is constructed using many FRAME pallets that ship with the
-[core Substrate repository](https://github.com/paritytech/substrate/tree/master/frame) and a
-template pallet that is [defined in the `pallets`](./pallets/template/src/lib.rs) directory.
+The runtime in this project is constructed using many FRAME pallets that ship with the [core Substrate repository](https://github.com/paritytech/substrate/tree/master/frame) and a template pallet that is [defined in the `pallets`](./pallets/template/src/lib.rs) directory.
 
 A FRAME pallet is compromised of a number of blockchain primitives:
 
-- Storage: FRAME defines a rich set of powerful
-  [storage abstractions](https://docs.substrate.io/main-docs/build/runtime-storage/) that makes
-  it easy to use Substrate's efficient key-value database to manage the evolving state of a
-  blockchain.
-- Dispatchables: FRAME pallets define special types of functions that can be invoked (dispatched)
-  from outside of the runtime in order to update its state.
-- Events: Substrate uses [events and errors](https://docs.substrate.io/main-docs/build/events-errors/)
-  to notify users of important changes in the runtime.
+- Storage: FRAME defines a rich set of powerful [storage abstractions](https://docs.substrate.io/build/runtime-storage/) that makes it easy to use Substrate's efficient key-value database to manage the evolving state of a blockchain.
+- Dispatchables: FRAME pallets define special types of functions that can be invoked (dispatched) from outside of the runtime in order to update its state.
+- Events: Substrate uses [events and errors](https://docs.substrate.io/build/events-and-errors/) to notify users of important changes in the runtime.
 - Errors: When a dispatchable fails, it returns an error.
-- Config: The `Config` configuration interface is used to define the types and parameters upon
-  which a FRAME pallet depends.
+- Config: The `Config` configuration interface is used to define the types and parameters upon which a FRAME pallet depends.
+
+## Alternatives Installations
+
+Instead of installing dependencies and building this source directly, consider the following alternatives.
 
-### Run in Docker
+### Nix
 
-First, install [Docker](https://docs.docker.com/get-docker/) and
-[Docker Compose](https://docs.docker.com/compose/install/).
+Install [nix](https://nixos.org/), and optionally [direnv](https://github.com/direnv/direnv) and [lorri](https://github.com/nix-community/lorri) for a fully plug-and-play experience for setting up the development environment.
+To get all the correct dependencies, activate direnv `direnv allow` and lorri `lorri shell`.
+
+### Docker
+
+First, install [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/).
 
 Then run the following command to start a single node development chain.
 
-```bash
+```sh
 ./scripts/docker_run.sh
```

Comment:
if i run clone the substrate repo, and check out this branch 'ds/template-readme', and then as root user i run `./bin/node-template/scripts/docker_run.sh` it outputs `-bash: ./bin/node-template/scripts/docker_run.sh: Permission denied`. if i run 
the following first `chmod +x ./bin/node-template/scripts/docker_run.sh` then i'm able to run that script.

if i then run `git ls-tree -r HEAD`, it outputs the following `100644 blob 0bac44b4cfb3b61959bc65ae6de72c22fade7ff4	bin/node-template/scripts/docker_run.sh`.

so to change its permissions to 755 i think we should run `git update-index --chmod=+x ./bin/node-template/scripts/docker_run.sh` and then commit that change to this PR?